app/src/main/java/ch/epfl/sweng/studdybuddy/FirebaseReference.java

```diff
@@ -0,0 +1,107 @@
+package ch.epfl.sweng.studdybuddy;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresPermission;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FirebaseReference implements ReferenceWrapper {
+    //pass it
+    /**
+     * Firebase reference implementation
+     */
+
+    //Reference to firebase
+    private final DatabaseReference ref;
+
+    public FirebaseReference() {
+        ref = FirebaseDatabase.getInstance().getReference();
+    }
+
+    public FirebaseReference(DatabaseReference firebaseRef) {
+        this.ref = firebaseRef;
+    }
+
+    public DatabaseReference getRef() { return ref; }
+
+    @Override
+    public ReferenceWrapper select(String key) {
+        return new FirebaseReference(ref.child(key));
+    }
+
+    @Override
+    public Task<Void> setVal(Object o) {
+        return ref.setValue(o);
+    }
+
+    @Override
+    public Task<Void> clear(){
+        return ref.removeValue();
+    }
+
+    @Override
+    public <T> void get(final Class<T> type, final Consumer<T> callback) {
+        ref.addValueEventListener(getValueEventListener(type, callback));
+    }
+
+    private <T> ValueEventListener getAllValueEventListener(final Class<T> type, final Consumer<List<T>> callback) {
+        return new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                List<T> elements = new ArrayList<>();
+                for(DataSnapshot snap: dataSnapshot.getChildren())
```

Comment:
A good practise is always adding brackets in if and for statements, even if they are one-liners.